porcupine/plugins/comment_block.py

```diff
@@ -32,17 +33,19 @@ def comment_or_uncomment(tab: tabs.FileTab, pressed_key: str | None = None) -> s
     all_linenos = set(range(start, end))
     commented = {
         lineno
-        for lineno in all_linenos
-        if tab.textwidget.get(f"{lineno}.0") == comment_prefix
-        # Do not touch comments starting with '# '
-        and tab.textwidget.get(f"{lineno}.1") != " "
+        for lineno, line in enumerate(
+            tab.textwidget.get(f"{start}.0", f"{end}.0").splitlines(), start
+        )
+        # Don't ignore indented '#    blah', that is most likely by this plugin
+        # But ignore '# blah' comments because they are likely written by hand
```

Comment:
```suggestion
        # Ignore '# blah' comments because they are likely written by hand
        # But don't ignore indented '#    blah', that is most likely by this plugin
```